delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/Storages.scala

```diff
@@ -751,4 +751,17 @@ object Storages {
     }
   }
 
+  /**
+    * Create an instance of [[EventExchange]] for [[StorageEvent]].
+    * @param storages storages operation bundle
+    */
+  def eventExchange(storages: Storages)(implicit crypto: Crypto, resolution: RemoteContextResolution): EventExchange =
+    EventExchange.create(
+      (event: StorageEvent) => storages.fetch(IriSegment(event.id), event.project).leftWiden[StorageRejection],
+      (event: StorageEvent, tag: TagLabel) =>
+        storages.fetchBy(IriSegment(event.id), event.project, tag).leftWiden[StorageRejection],
+      (storage: Storage) => storage.toExpandedJsonLd,
+      (storage: Storage) => UIO.pure(Storage.encryptSource(storage.source, crypto).getOrElse(Json.Null))
```

Comment:
maybe `Json.obj()` is more appropriate instead of `Json.Null`?

Reply:
or alternately returning an error?

Reply:
This should issue an error, yes